### PR TITLE
Instead of maintaining cache id etc, reset the "working" exports map cache.

### PR DESCRIPTION
**Pull request by sheetalkamat**
Is this better alternative to #45238

### CODE REVIEW

**Comment by sheetalkamat** [comment]
@amcasey can you check if this is helpful?

**Comment by amcasey** [comment]
My notes from the time: https://github.com/microsoft/TypeScript/issues/45283

File: src/compiler/builder.ts [hunk 1]
```diff
@@ -372,6 +372,7 @@ namespace ts {
                 BuilderState.updateSignaturesFromCache(state, state.currentAffectedFilesSignatures!);
                 state.currentAffectedFilesSignatures!.clear();
                 BuilderState.updateExportedFilesMapFromCache(state, state.currentAffectedFilesExportedModulesMap);
+                state.currentAffectedFilesExportedModulesMap = undefined;
```
Comment (sheetalkamat) [inline src/compiler/builder.ts:375]: I think this got missed when we first introduced `currentAffectedFilesExportedModulesMap`.. Look how we clear `currentAffectedFilesSignatures` just above..

File: src/compiler/builder.ts [hunk 2]
```diff
@@ -372,6 +372,7 @@ namespace ts {
                 BuilderState.updateSignaturesFromCache(state, state.currentAffectedFilesSignatures!);
                 state.currentAffectedFilesSignatures!.clear();
                 BuilderState.updateExportedFilesMapFromCache(state, state.currentAffectedFilesExportedModulesMap);
+                state.currentAffectedFilesExportedModulesMap?.clear();
```
Comment (sheetalkamat) [inline src/compiler/builder.ts:375]: I think this got missed when we first introduced currentAffectedFilesExportedModulesMap.. Look how we clear currentAffectedFilesSignatures just above..

**Comment by amcasey** [comment]
For a single run of each, I see a 3% regression in total time, mostly in emit.  Memory usage was also quite a bit higher, but that tends to fluctuate.

**Review by amcasey — APPROVED** [review]
Perf looks comparable on the project that motivated the change.

**Pull request merged**